library/core/src/ptr/const_ptr.rs

```diff
@@ -1438,6 +1438,37 @@ impl<T: ?Sized> *const T {
         self.is_aligned_to(align_of::<T>())
     }
 
+    /// Returns whether the pointer is properly aligned for `U`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(pointer_is_aligned_for)]
+    /// 
+    /// // On some platforms, the alignment of i32 is less than 4.
+    /// #[repr(align(4))]
+    /// struct AlignedI32(i32);
+    /// 
+    /// // On some platforms, the alignment of u32 is less than 4.
+    /// #[repr(align(4))]
+    /// struct AlignedU32(u32);
+    ///
+    /// let data = AlignedI32(42);
+    /// let ptr = &data as *const AlignedI32;
+    ///
+    /// assert!(ptr.is_aligned_for::<AlignedU32>());
+    /// assert!(!ptr.wrapping_byte_add(1).is_aligned_for::<AlignedU32>());
+    /// ```
+    #[must_use]
+    #[inline]
+    #[unstable(feature = "pointer_is_aligned_for", issue = "140980")]
+    pub fn is_aligned_for<U: Sized>(self) -> bool
+    where
+        T: Sized,
```

Comment:
Is there a reason to restrict these to `T: Sized`? The only alignment that matters is of `U`, so we don't need to know `T`'s alignment. Similar to `<*const T>::cast`, which only requires `U: Sized` (If `T` is unsized, `cast` just discards the pointer metadata).